language/tools/test_generation/tests/common.rs

```diff
@@ -3,20 +3,16 @@ use test_generation::{abstract_state::AbstractState, summaries::instruction_summ
 use vm::file_format::Bytecode;
 
 pub fn run_instruction(instruction: Bytecode, initial_state: AbstractState) -> AbstractState {
-    let instruction_summary = instruction_summary(instruction);
-    let mut preconditions_satisfied = true;
-    for precondition in instruction_summary.preconditions.into_iter() {
-        if !precondition(&initial_state) {
-            preconditions_satisfied = false;
-        }
-    }
+    let summary = instruction_summary(instruction);
+    let unsatisfied_preconditions = summary
+        .preconditions
+        .iter()
+        .any(|precondition| !precondition(&initial_state));
     assert_eq!(
```

Comment:
This looks like a place to use checked_assume!

Reply:
Actually, this function is used both for tests that are supposed to fail or supposed to succeed. In the cases where they are supposed to fail we would not want to assume that all the preconditions are satisfied.